Try GitHub pull request trigger actions

christianhueserhzdr opened the pull request:
Changed GitHub action events which trigger pipelines in LabHub TOML file

The actions which trigger pipelines are 
* synchronize
* opened
* reopened
* closed

Please check if that is correct.

All possible actions are listed here:
https://developer.github.com/v3/activity/events/types/#pullrequestevent

christianhueserhzdr committed:
Add more pull request trigger actions

Added "opened", "reopened", "closed" (merged) as pull request trigger actions.

christianhueserhzdr committed:
Remove some unneccessary pull request trigger actions

Not all trigger actions are relevant. Removed some but "synchronized",
"opened", "reopend".

christianhueserhzdr committed:
Simplify check for pull request trigger actions

Check action "merged" instead of action "closed".

christianhueserhzdr committed:
Re-enable "closed" trigger action

Check if key "merged" is true if action is "closed".

christianhueserhzdr committed:
Replaced "merged" by "closed" pull request trigger action

There is no pull request trigger action called "merged". It is included
in "closed".

tobiashuste commented on src/actions/actions.rs:
```diff
@@ -2,8 +2,10 @@
     Synchronize,
+    Opened,
+    Reopened,
+    Closed,
```
One question: in case somebody defines non-default configuration options in his `LabHub.toml`, would this still work or do all possible options need to be mentioned here?

christianhueserhzdr replied:
Yes, it will still work. The TOML file may contain lots of entries, but only those which are actual actions will be taken into account.

christianhueserhzdr replied:
Ok, I see what you mean. It might be reasonable to add the complete list since this is no config file.

tobiashuste commented on LabHub.toml:
```diff
@@ -40,6 +40,8 @@ gitlab_repo = "brndnmtthws-oss/conky"
     "synchronize",
+    "opened",
+    "reopened",
+    "closed",
```
To me, this list looks totally fine. When we submit the PR to the original repo, it might be possible that they request us to add the complete list here in order to not cause a different behavior than before.

But let's wait for their review and see what they say.

christianhueserhzdr replied:
Alright, you are right. We need to put everything in here for the original repo.

tobiashuste replied:
But let's just wait for their reaction first. To me, this new behavior makes more sense.

tobiashuste reviewed (commented):
We should squash the commits before merging to master. Thanks, for working on this :)

christianhueserhzdr committed:
Add all possible options of github trigger actions

Configuration of enabled trigger actions need to be done in LabHub TOML
configuration file.

tobiashuste commented:
@christianhueserhzdr I rebased to the current master. Can you squash and merge, please?
